contracts/rewards/boosted-staking/BoostedVault.sol

```diff
@@ -249,6 +250,45 @@ contract BoostedVault is
         _withdraw(_amount);
     }
 
+    /**
+     * @dev Withdraws given stake amount from the pool and
+     * redeems the staking token into a given asset.
+     * @param _amount        Units of the staked token to withdraw
+     * @param _minAmountOut  TODO
+     * @param _output        TODO
+     * @param _beneficiary   Address to send staked token to
+     * @param _router        Router to redeem/swap
+     * @param _isBassetOut     Route action of redeem/swap
+     */
+    function withdrawAndUnwrap(
+        uint256 _amount,
+        uint256 _minAmountOut,
+        address _output,
+        address _beneficiary,
+        address _router,
+        bool _isBassetOut
+    ) external override updateReward(msg.sender) updateBoost(msg.sender) {
+        require(_amount > 0, "Cannot withdraw 0");
+
+        // Reduce raw balance (but do not transfer `stakingToken`)
+        _reduceRaw(_amount);
+
+        // Approve SavingsContract to spend this contract's `stakingToken`
+        IERC20(stakingToken).approve(address(stakingToken), _amount);
```

Comment:
Do we need to approve here? I think the savingscontract can just burn the tokens from msg.sender (which will be this staking contract account when called below)